consensus/src/round_manager.rs

```diff
@@ -1222,6 +1224,47 @@ impl RoundManager {
         result
     }
 
+    async fn new_qc_from_order_vote_msg(
+        &mut self,
+        order_vote_msg: &OrderVoteMsg,
+    ) -> anyhow::Result<()> {
+        ensure!(
+            order_vote_msg
+                .order_vote()
+                .ledger_info()
+                .consensus_block_id()
+                == order_vote_msg.quorum_cert().certified_block().id(),
+            "QuorumCert attached to order vote doesn't match"
+        );
+
+        if let NeedFetchResult::QCAlreadyExist = self
+            .block_store
+            .need_fetch_for_quorum_cert(order_vote_msg.quorum_cert())
+        {
+            return Ok(());
+        }
+
+        let start = Instant::now();
+        order_vote_msg
+            .quorum_cert()
+            .verify(&self.epoch_state().verifier)
+            .context("[OrderVoteMsg QuorumCert verification failed")?;
+        counters::VERIFY_MSG
+            .with_label_values(&["order_vote_qc"])
+            .observe(start.elapsed().as_secs_f64());
+
+        let result = self
+            .block_store
+            .insert_quorum_cert(
```

Comment:
this has implicit block fetch if the block not exist, not sure if we want to go down this path. today we only fetch it when the order vote cert aggregated, after the pr it'll fetch when the first order vote comes